### ISSUE TITLE
Load balancing https return codes

### ISSUE THREAD

**Issue by rwidmark** [ironmansoftware/powershell-universal#1573]
### Summary of the new feature / enhancement

So this feature is something that many of us want.

we are using multiple psu servers and load balancing with ADC.

so what we need is something to check against so https check with the following return codes:

200 - OK
503 - maintaince 
500 - server is down

It would also be perfect if it was possible to limit the amount of users to scale the servers.
For example if the amount of users reach over 500 people then return;
HTTPS return code 302

**Comment by rwidmark** [comment]
This is badly needed @adamdriscoll

**Comment by adamdriscoll** [comment]
I'm curious what the server down error would be from. I'm thinking:

- Dashboards failed to start
- API\Security proxies failed to start

If the server is down, down then it won't respond at all but we can certainly return a 500 if certain states exist.

**Comment by rwidmark** [comment]
> I'm curious what the server down error would be from. I'm thinking:
> 
> 
> 
> - Dashboards failed to start
> 
> - API\Security proxies failed to start
> 
> 
> 
> If the server is down, down then it won't respond at all but we can certainly return a 500 if certain states exist. 

@adamdriscoll I'm guessing you need two 500 codes then :)
Dident think of that.

**Comment by Viajaz** [comment]
I must admit, this is on my list of roadblockers regarding our expansion our use of PowerShell Universal.

**Comment by rwidmark** [comment]
@Viajaz we have done some work a rounds but I agree with you that this is a issue in production.

What I have done now is that the ADC are looking at port 443 is it's alive.

Then I have set the Client-idle timeout value to the same that I have one the dashboard.

I have also turned off Multiplexing as that will mess things up if it's on as you want it to speak with the same server during the hole session.

And as I don't know the sizing for powershell universal I have set the maximum for each server at 500  clients.

@adamdriscoll do you have any sizing info?

**Comment by Viajaz** [comment]
@rstolpe I had some PSU API performance issues a few months ago and @adamdriscoll ran his benchmarking suite against a basic PSU API Endpoint PowerShell script I came up with for stress testing CPU, CPU+IO CPU+SQL etc. and got a average requests per-second of ~536 against just the CPU-bound test (Just hash computing) although a profiling breakdown showed the overhead consumed most of the time in my very basic example. Adam got ~1082 in April against an empty API endpoint with Siege. Unfortunately, I don't know what hardware resources Adam was throwing at those benchmarks.

Some useful links:
- https://docs.powershelluniversal.com/development/profiling
- https://blog.ironmansoftware.com/webapp-benchmark-siege/

**Comment by adamdriscoll** [comment]
Those results were based on a machine with 8 vCPU cores and 16 GB memory. It wasn't using a SQL backend but rather the LiteDB storage. That said, it was mostly just testing API throughput and not storing much data. 

This issue has been resolved. There is a new api/v1/status endpoint that will return 200 if the server started successfully and all dashboards are running. Any configuration issues will result in a 500 error code. We've also added maintenance mode for computers and then the status page will return 503.
